### PR TITLE
Add support for MiniMapCanFlip

### PR DESCRIPTION
**Pull request by badosu**
Remarks:

- `Gadgets/unit_cloak_shield.lua` callin for `DrawInMiniMap` seems to be noop. Disabling the gadget does not change anything, the range is still drawn correctly, flipped or not.

Untested widgets:

- `Widgets/minimap_events.lua`

These should be safe when minimap cant flip or is not flipped, otherwise they follow same structure for drawing as other tested ones.

All changes, unless stated otherwise above, were tested.

### CODE REVIEW

File: LuaUI/Headers/minimap_utilities.lua [hunk 1]
```diff
@@ -0,0 +1,12 @@
+local function getMiniMapFlipped()
+	if ((not Spring.GetMiniMapRotation) or
+	    (Spring.GetConfigInt("MiniMapCanFlip", 0) == 0)) then
```
Comment (sprunk) [inline LuaUI/Headers/minimap_utilities.lua:3]: Is the config check needed? I think if this config value is set to false then `Spring.GetMiniMapRotation` should always return 0.

Comment (badosu) [reply LuaUI/Headers/minimap_utilities.lua:3]: Not necessary probably, I just like to decouple from specific implementation cases

Comment (badosu) [reply LuaUI/Headers/minimap_utilities.lua:3]: If you send the suggestion I can incorporate it

**Comment by sprunk** [comment]
`minimap_startbox_new.lua` seems not to work. I think this is because of `gl.CallList(boxMinimapList)` which is above the flip check and transforms.
screen00055
screen00056

**Comment by sprunk** [comment]
Here's the startscript I use if you need to reproduce (drag'n'drop onto spring.exe, should work with any other map as well) [script.txt](https://github.com/ZeroK-RTS/Zero-K/files/10200375/script_pentos.txt)

**Comment by badosu** [comment]
Thank you! Let me reproduce

**Comment by sprunk** [comment]
@badosu have you done anything in this area? I'd like to look into it myself but wouldn't want to duplicate work.

**Comment by badosu** [comment]
I'm trying to run Zero-K from skylobby for the startbox start but it seems to not work (it can't find "Zero-K $VERSION").

If you know how to test this scenario running directly from the engine binary I can address it.

**Comment by badosu** [comment]
@sprunk This should be addressed now

**Comment by badosu** [comment]
I'm stupid, didn't check again that you actually shared the startscript previously.

**Comment by sprunk** [comment]
Thanks!

**Pull request merged**